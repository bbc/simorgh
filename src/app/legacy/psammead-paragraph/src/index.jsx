import styled from '@emotion/styled';
import { shape, string, bool } from 'prop-types';
import { C_LUNAR, C_GREY_10 } from '#legacy/psammead-styles/src/colours';
import { GEL_SPACING_TRPL } from '#legacy/gel-foundations/src/spacings';
import { getBodyCopy } from '#legacy/gel-foundations/src/typography';
import { scriptPropType } from '#legacy/gel-foundations/src/prop-types';
import { getSansRegular } from '#legacy/psammead-styles/src/font-styles';

const Paragraph = styled.p`
  ${({ script }) => script && getBodyCopy(script)};
  ${({ service }) => getSansRegular(service)}
  color: ${({ darkMode }) => (darkMode ? C_LUNAR : C_GREY_10)};
  padding-bottom: ${GEL_SPACING_TRPL};
  margin: 0; /* Reset */
`;

Paragraph.propTypes = {
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
  darkMode: bool,
};

Paragraph.defaultProps = {
  darkMode: false,
};

export default Paragraph;
