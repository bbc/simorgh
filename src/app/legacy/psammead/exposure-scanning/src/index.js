import parseArgs from './args';
import { fetchPr, fetchIssue } from './fetch';
import { scanPr, scanIssue } from './scan';
import { patchPr, patchIssue } from './patch';

const redactPr = async (reqBody, regex) => {
  const pr = await fetchPr(reqBody);

  const scannedPr = await scanPr(pr, regex);

  if (scannedPr.foundMatch) {
    await patchPr(reqBody, scannedPr);
    throw new Error('Match found and patched.');
  }
};

const redactIssue = async (reqBody, regex) => {
  const issue = await fetchIssue(reqBody);

  const scannedIssue = await scanIssue(issue, regex);

  if (scannedIssue.foundMatch) {
    await patchIssue(reqBody, scannedIssue);
    throw new Error('Match found and patched.');
  }
};

const scanExposures = async () => {
  const { repo, flag, id, regexString } = parseArgs(process.argv);

  const regex = new RegExp(regexString, 'gi');

  const reqBody = {
    owner: 'bbc',
    repo,
    id,
  };

  await {
    '-pr': redactPr,
    '-issue': redactIssue,
  }[flag](reqBody, regex);

  return 'No matches found.';
};

export default scanExposures;
