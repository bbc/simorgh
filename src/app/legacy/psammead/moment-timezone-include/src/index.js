const webpack = require('webpack');
const writeNewTimezoneData = require('./writeNewTimezoneData');
const replaceOriginalTimezoneData = require('./replaceOriginalTimezoneData');

function MomentTimezoneInclude(options = {}) {
  const startYear = options.startYear || -Infinity;
  const endYear = options.endYear || Infinity;

  writeNewTimezoneData(startYear, endYear);

  return new webpack.NormalModuleReplacementPlugin(
    /data[\\/]packed[\\/]latest\.json$/,
    resource => {
      resource.request = replaceOriginalTimezoneData(); // eslint-disable-line no-param-reassign
    },
  );
}

module.exports = MomentTimezoneInclude;
