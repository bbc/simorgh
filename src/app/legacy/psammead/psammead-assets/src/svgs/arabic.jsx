import React from 'react';

const height = 24.59;
const width = 222.3514;
const arabic = {
  group: (
    <g
      fillRule="evenodd"
      stroke="#000"
      strokeWidth=".335"
      style={{ stroke: '#fff' }}
    >
      <path d="M76.73.28v21.85H53.51V.28zm-4.62 14.58l-.37.23a10.85 10.85 0 0 1-5.53 1.69c-3.8 0-6.31-2.27-6.32-5.55s2.62-5.58 6.22-5.59a10.93 10.93 0 0 1 5.47 1.56l.36.2V4.47l-.16-.06a15.06 15.06 0 0 0-5.65-1.27 9.62 9.62 0 0 0-6.53 2.34 7.9 7.9 0 0 0-2.59 6 7.7 7.7 0 0 0 2.16 5.17A9.1 9.1 0 0 0 66 19.28a12.73 12.73 0 0 0 6-1.37l.14-.07zM50 .28v21.85H26.75V.28zm-5.54 14.48A4.21 4.21 0 0 0 41 10.69a3.87 3.87 0 0 0 1.35-1.08A3.41 3.41 0 0 0 43 7.47a3.8 3.8 0 0 0-1.27-2.85 5.76 5.76 0 0 0-4-1.29h-4.89v15.75h5.81a6.54 6.54 0 0 0 4.46-1.39 3.88 3.88 0 0 0 1.35-2.93z" />
      <path d="M41.57 14.45a1.86 1.86 0 0 1-.63 1.45 3.85 3.85 0 0 1-2.6.72h-2.68v-4.3h2.56a4.45 4.45 0 0 1 2.57.62 1.76 1.76 0 0 1 .78 1.51zM39.4 9.31a1.84 1.84 0 0 0 .74-1.63 1.66 1.66 0 0 0-.53-1.31 3 3 0 0 0-2.05-.57h-1.9v4.07H37a4.21 4.21 0 0 0 2.4-.56zM23.22.28v21.85H0V.28zM17.7 14.76a4.21 4.21 0 0 0-3.43-4.07 3.86 3.86 0 0 0 1.35-1.08 3.41 3.41 0 0 0 .65-2.13A3.8 3.8 0 0 0 15 4.63a5.76 5.76 0 0 0-4-1.29H6v15.74h5.9a6.54 6.54 0 0 0 4.46-1.39 3.89 3.89 0 0 0 1.34-2.93z" />
      <path d="M12.64 9.31a1.85 1.85 0 0 0 .74-1.63 1.66 1.66 0 0 0-.53-1.31 3 3 0 0 0-2.05-.57H8.91v4.07h1.3a4.21 4.21 0 0 0 2.43-.56zM14 12.94a4.45 4.45 0 0 0-2.57-.62H8.91v4.29h2.68a3.85 3.85 0 0 0 2.6-.72 1.86 1.86 0 0 0 .63-1.45 1.76 1.76 0 0 0-.82-1.5zM101 3.34h2.15v15.78h-1.94L90.69 7v12.12h-2.12V3.34h1.83L101 15.6zM107.26 3.34h8.95v2h-6.69v4.81H116v2h-6.46v4.9h6.9v2h-9.16zM140.33 3.34h2.25l-6.38 15.85h-.5l-5.16-12.84-5.21 12.84h-.49l-6.36-15.85h2.28l4.35 10.88 4.38-10.88h2.14L136 14.22zM148.7 12.51l-1.72-1a7.8 7.8 0 0 1-2.3-1.94 3.68 3.68 0 0 1-.68-2.2 3.88 3.88 0 0 1 1.29-3 4.83 4.83 0 0 1 3.36-1.16 6.36 6.36 0 0 1 3.63 1.11v2.49a5.23 5.23 0 0 0-3.67-1.64 3 3 0 0 0-1.82.51 1.55 1.55 0 0 0-.71 1.32 2 2 0 0 0 .52 1.33 6.6 6.6 0 0 0 1.69 1.3l1.73 1q2.89 1.73 2.89 4.39a4 4 0 0 1-1.27 3.08 4.65 4.65 0 0 1-3.3 1.19 6.94 6.94 0 0 1-4.26-1.44V15a5.32 5.32 0 0 0 4.24 2.32 2.66 2.66 0 0 0 1.77-.59 1.85 1.85 0 0 0 .71-1.48q-.02-1.45-2.1-2.74z" />
      <path d="M168.37 0v22.59" strokeWidth="2" />
      <path d="M203 4.64c-.11-.21-.26-.49-.32-.58h-.06L201 5.56v.06l.13.19.17.26a.79.79 0 0 0 .1.14 6.14 6.14 0 0 1 1.08 3.66v.25h-10.4l.42 1.76c0 .13 0 .2.3.2h2.86l-.16.35a4.36 4.36 0 0 1-4.26 2.7c-2.49 0-4.52-2-5.44-5.3l-.8 1.77a8.5 8.5 0 0 0 1.07 2.64 6.42 6.42 0 0 0 5.6 2.9 6.53 6.53 0 0 0 6.06-4.82v-.19h6.43c.11-1.07.16-1.53.16-1.92a6.12 6.12 0 0 0 0-.62A10.53 10.53 0 0 0 203 4.64M200.35 15.23a.27.27 0 0 0-.06.16.22.22 0 0 0 .05.13l1 1a.28.28 0 0 0 .16.06s.07 0 .16-.08l1-1a.24.24 0 0 0 .08-.13.21.21 0 0 0-.08-.16l-1-1a.18.18 0 0 0-.13-.06.25.25 0 0 0-.16.06zM190 18.36a.24.24 0 0 0-.16.06l-1 1a.18.18 0 0 0-.06.14.3.3 0 0 0 .08.19l1 1s.06.06.16.06a.18.18 0 0 0 .13-.06l1-1a.23.23 0 0 0 .05-.13.31.31 0 0 0-.06-.16l-1-1a.29.29 0 0 0-.15-.07" />
      <path d="M192.4 18.36a.3.3 0 0 0-.19.08l-1 1a.14.14 0 0 0-.06.14.3.3 0 0 0 .08.19l1 1a.27.27 0 0 0 .16.06h.11l1-1a.23.23 0 0 0 .08-.13.25.25 0 0 0-.08-.16l-1-1a.18.18 0 0 0-.13-.06M218.85 10.2a2.28 2.28 0 0 1-2.5-2.37c0-1.28.68-2.34 3.92-2.34h.55V4.31a13.17 13.17 0 0 0-2.49-.24c-2.43 0-4.2 1.45-4.2 3.45a3.72 3.72 0 0 0 .71 2.29l.26.39h-4.52v-.22a16.31 16.31 0 0 0-.84-3.62c-.14-.4-.34-1-.45-1.2l-1.7 1.55a.24.24 0 0 0 0 .06.71.71 0 0 0 .12.32 8.9 8.9 0 0 1 1.14 4.41 4.44 4.44 0 0 1-.55 2.37 4.27 4.27 0 0 1-3.3 1.83l1.49 1.42c2.52-.4 4-2.24 4.2-5.05a.78.78 0 0 0 .37.08h10.64c.21 0 .32-.06.32-.16V10.2z" />
    </g>
  ),
  ratio: width / height,
  viewbox: {
    height,
    width,
  },
};

export default arabic;
