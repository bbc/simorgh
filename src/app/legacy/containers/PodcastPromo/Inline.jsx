import React, { useContext } from 'react';
import styled from '@emotion/styled';
import pathOr from 'ramda/src/pathOr';
import {
  GEL_SPACING_HLF,
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_SPACING_TRPL,
} from '#psammead/gel-foundations/src/spacings';
import {
  GEL_GROUP_0_SCREEN_WIDTH_MAX,
  GEL_GROUP_1_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';
import {
  getPica,
  getBrevier,
  getLongPrimer,
} from '#psammead/gel-foundations/src/typography';
import { getSerifMedium } from '#psammead/psammead-styles/src/font-styles';
import useViewTracker from '#hooks/useViewTracker';
import useClickTrackerHandler from '#hooks/useClickTrackerHandler';

import ImageWithPlaceholder from '#containers/ImageWithPlaceholder';
import SkipLinkWrapper from '#components/SkipLinkWrapper';
import { ServiceContext } from '../../../contexts/ServiceContext';
import PromoComponent from './components';
import getPromo from './shared';

const GEL_GROUP_1_WIDTH_260PX = '16.25rem';
const GEL_GROUP_1_WIDTH_320PX = '20rem';
const GEL_GROUP_1_WIDTH_360PX = '22.5rem';

const ResponsivePodcastPromoWrapper = styled.div`
  ${({ dir }) => (dir === 'ltr' ? 'float: right;' : 'float: left;')}
  background: ${props => props.theme.palette.LUNAR};
  margin: ${GEL_SPACING_TRPL} 0;
  height: auto;

  @media (min-width: ${GEL_GROUP_1_SCREEN_WIDTH_MIN}) {
    width: 45%;
    margin: ${GEL_SPACING_TRPL} ${GEL_SPACING_HLF};
  }

  @media (min-width: ${GEL_GROUP_1_WIDTH_260PX}) {
    margin: ${GEL_SPACING_TRPL} ${GEL_SPACING};
  }

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    margin: ${GEL_SPACING_TRPL} ${GEL_SPACING_DBL};
  }
`;

const StyledPromoComponent = styled(PromoComponent)`
  padding: ${GEL_SPACING_DBL} ${GEL_SPACING} ${GEL_SPACING} ${GEL_SPACING};
`;

const StyledImageWrapper = styled(PromoComponent.Card.ImageWrapper)`
  display: inline-block;
  width: 100%;
  margin: 0;
  padding: ${GEL_SPACING};

  @media (max-width: ${GEL_GROUP_0_SCREEN_WIDTH_MAX}) {
    display: none;
  }
`;

const StyledCardContentWrapper = styled(PromoComponent.Card.Content)`
  @media (max-width: ${GEL_GROUP_0_SCREEN_WIDTH_MAX}) {
    padding: ${GEL_SPACING};
  }

  @media (min-width: ${GEL_GROUP_1_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_SPACING} ${GEL_SPACING} ${GEL_SPACING};
  }
`;

const StyledCardDescriptionWrapper = styled(PromoComponent.Card.Description)`
  margin: ${GEL_SPACING} 0;
  overflow-wrap: break-word;
`;

const StyledEpisodeTextWrapper = styled(PromoComponent.Card.EpisodesText)`
  ${({ script }) => getBrevier(script)}
  @media (max-width: ${GEL_GROUP_0_SCREEN_WIDTH_MAX}) {
    margin: 0 ${GEL_SPACING_HLF};
  }

  @media (min-width: ${GEL_GROUP_1_SCREEN_WIDTH_MIN}) {
    margin: 0;
  }

  @media (min-width: ${GEL_GROUP_1_WIDTH_260PX}) {
    margin: 0;
  }

  @media (min-width: ${GEL_GROUP_1_WIDTH_320PX}) {
    margin: 0 ${GEL_SPACING_HLF};
  }

  @media (min-width: ${GEL_GROUP_1_WIDTH_360PX}) {
    margin: 0;
  }

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    ${({ script }) => getLongPrimer(script)}
    margin: 0 ${GEL_SPACING_HLF};
  }
`;

const StyledCardLink = styled(PromoComponent.Card.Link)`
  ${({ script }) => getLongPrimer(script)}
  ${({ service }) => getSerifMedium(service)}
  display: block;

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    ${({ script }) => getPica(script)}
  }
`;

const Promo = () => {
  const { podcastPromo, script, service, dir } = useContext(ServiceContext);

  const {
    podcastPromoTitle,
    podcastBrandTitle,
    description,
    imgSrc,
    alt,
    url,
    label,
    showPromo,
    eventTrackingData,
    sizes,
    srcset,
    primaryMimeType,
  } = getPromo(podcastPromo);

  const viewTrackerRef = useViewTracker(eventTrackingData);
  const clickTrackerRef = useClickTrackerHandler(eventTrackingData);

  if (!showPromo) {
    return null;
  }

  const { text, endTextVisuallyHidden } = pathOr(
    {
      text: 'Skip %title% and continue reading',
      endTextVisuallyHidden: 'End of %title%',
    },
    ['skipLink'],
    podcastPromo,
  );

  const terms = {
    '%title%': podcastPromoTitle,
  };

  return (
    <ResponsivePodcastPromoWrapper ref={viewTrackerRef} dir={dir}>
      <StyledPromoComponent
        script={script}
        service={service}
        role="region"
        aria-labelledby="podcast-promo"
      >
        <SkipLinkWrapper
          endTextId="end-of-podcasts"
          terms={terms}
          text={text}
          endTextVisuallyHidden={endTextVisuallyHidden}
          service={service}
        >
          <PromoComponent.Title id="podcast-promo" dir={dir} as="strong">
            {podcastPromoTitle}
          </PromoComponent.Title>
          <PromoComponent.Card inlinePromo>
            <StyledImageWrapper>
              <ImageWithPlaceholder
                src={imgSrc}
                srcset={srcset}
                primaryMimeType={primaryMimeType}
                sizes={sizes}
                alt={alt}
                height={1}
                width={1}
                ratio={100}
                lazyLoad
              />
            </StyledImageWrapper>
            <StyledCardContentWrapper>
              <StyledCardLink
                href={url}
                onClick={clickTrackerRef}
                script={script}
                service={service}
              >
                <span className="podcast-promo--hover podcast-promo--focus podcast-promo--visited">
                  {podcastBrandTitle}
                </span>
              </StyledCardLink>
              <StyledCardDescriptionWrapper>
                {description}
              </StyledCardDescriptionWrapper>
              <StyledEpisodeTextWrapper dir={dir} script={script}>
                {label}
              </StyledEpisodeTextWrapper>
            </StyledCardContentWrapper>
          </PromoComponent.Card>
        </SkipLinkWrapper>
      </StyledPromoComponent>
    </ResponsivePodcastPromoWrapper>
  );
};

export default Promo;
