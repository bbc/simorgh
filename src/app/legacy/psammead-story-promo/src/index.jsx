import React from 'react';
import styled from '@emotion/styled';
import { node, bool, string, oneOf, shape } from 'prop-types';
import {
  GEL_SPACING_HLF,
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_SPACING_QUAD,
} from '#legacy/gel-foundations/src/spacings';
import {
  GEL_GROUP_1_SCREEN_WIDTH_MAX,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_2_SCREEN_WIDTH_MAX,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MAX,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
  GEL_GROUP_5_SCREEN_WIDTH_MIN,
} from '#legacy/gel-foundations/src/breakpoints';
import {
  getParagon,
  getLongPrimer,
  getPica,
  getDoublePica,
} from '#legacy/gel-foundations/src/typography';
import {
  C_EBON,
  C_GREY_10,
  C_METAL,
  C_SHADOW,
  C_WHITE,
} from '#legacy/psammead-styles/src/colours';
import {
  getSansRegular,
  getSerifMedium,
} from '#legacy/psammead-styles/src/font-styles';
import { scriptPropType } from '#legacy/gel-foundations/src/prop-types';
import { grid } from '#legacy/psammead-styles/src/detection';
import ImageGridItem from './ImageStyles';
import TextGridItem from './TextStyles';

const PROMO_TYPES = oneOf([
  'top',
  'regular',
  'leading',
  'topStories',
  'relatedContent',
]);

const wrapperTopStyles = `
  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    grid-template-columns: repeat(12, 1fr);
  }
`;

const wrapperRelatedContent = `
  background-color: ${C_WHITE};
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    display: block;
  };
`;

const wrapperTopStoriesStyle = ({ displayTimestamp }) => `
  background-color: ${C_WHITE};
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    display: block;
  };
  padding:  ${
    displayTimestamp
      ? GEL_SPACING
      : `${GEL_SPACING} ${GEL_SPACING} 0 ${GEL_SPACING}`
  };

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: ${
      displayTimestamp
        ? `${GEL_SPACING_DBL} ${GEL_SPACING} `
        : `${GEL_SPACING_DBL} ${GEL_SPACING} ${GEL_SPACING} ${GEL_SPACING}`
    };
  };

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    padding: ${
      displayTimestamp
        ? GEL_SPACING_DBL
        : `${GEL_SPACING_DBL} ${GEL_SPACING_DBL} ${GEL_SPACING} ${GEL_SPACING_DBL}`
    };
  };
`;

const wrapperRegularStyles = `
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    display: block;
  }
`;

const wrapperStyles = {
  top: wrapperTopStyles,
  regular: wrapperRegularStyles,
  topStories: wrapperTopStoriesStyle,
  relatedContent: wrapperRelatedContent,
  leading: '',
};

const StoryPromoWrapper = styled.div`
  position: relative; /* This is needed to contain the faux-block-link to the Story Promo */
  @supports (${grid}) {
    display: grid;
    grid-template-columns: repeat(6, 1fr);
    grid-column-gap: ${GEL_SPACING};

    @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
      grid-column-gap: ${GEL_SPACING_DBL};
    }
    ${({ promoType }) => wrapperStyles[promoType]}
  }
`;

const ImageContentsWrapper = styled.div`
  position: relative;
`;

const mediaIndicatorStylesTopLeading = `
  position: absolute;
  bottom: 0;
  > * {
    height: ${GEL_SPACING_QUAD};
    padding: ${GEL_SPACING} ${GEL_SPACING_HLF};
  }
`;

const mediaIndicatorStylesRegular = `
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    position: absolute;
    bottom: 0;
  }
  > * {
    @media (max-width: ${GEL_GROUP_1_SCREEN_WIDTH_MAX}) {
      height: 1.25rem;
      padding: ${GEL_SPACING_HLF} ${GEL_SPACING_HLF} 0;
    }

    height: ${GEL_SPACING_QUAD};
    padding: ${GEL_SPACING} ${GEL_SPACING_HLF};
  }
`;

/*
 These styles are to ensure we have the correct positioning
 & spacing of the Media Indicator over the Image in the Story Promo 
 */
const mediaIndicatorStyles = {
  top: mediaIndicatorStylesTopLeading,
  regular: mediaIndicatorStylesRegular,
  topStories: mediaIndicatorStylesRegular,
  relatedContent: mediaIndicatorStylesRegular,
  leading: mediaIndicatorStylesTopLeading,
};

const ImageOverlayWrapper = styled.div`
  ${({ promoType }) => mediaIndicatorStyles[promoType]}
`;

const headlineTopStoryTypography = script => getParagon(script);

const headlineRegularTypography = script => getPica(script);

const headlineLeadingStoryTypography = script => getDoublePica(script);

const headlineTypography = script => ({
  top: headlineTopStoryTypography(script),
  regular: headlineRegularTypography(script),
  topStories: headlineRegularTypography(script),
  relatedContent: headlineRegularTypography(script),
  leading: headlineLeadingStoryTypography(script),
});

const StandardColor = `
  color: ${C_EBON};
`;
const TopStoryColor = `
  color: ${C_GREY_10};
`;

const relatedContentStyle = `
  color: ${C_GREY_10};
  padding: 0;
`;
const headlineStyle = {
  top: StandardColor,
  regular: StandardColor,
  topStories: TopStoryColor,
  relatedContent: relatedContentStyle,
  leading: StandardColor,
};

export const Headline = styled.h3`
  margin: 0; /* Reset */
  padding-bottom: ${GEL_SPACING};
  ${({ service }) => getSerifMedium(service)}
  ${({ script, promoType }) => script && headlineTypography(script)[promoType]}
  ${({ promoHasImage }) =>
    !promoHasImage &&
    `display: inline-block;`} /* Needed for aligning Media Indicator with Headline */
  ${({ promoType }) => headlineStyle[promoType]}
`;

Headline.propTypes = {
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
  promoHasImage: bool,
  promoType: PROMO_TYPES,
};

Headline.defaultProps = {
  promoHasImage: true,
  promoType: 'regular',
};

export const RelatedContentBoxWrapper = styled.div`
  padding: ${GEL_SPACING};
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: ${GEL_SPACING_DBL};
  }
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    padding: ${GEL_SPACING} ${GEL_SPACING_DBL};
  }
`;

const summaryTopStoryStyles = `
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    display: none;
    visibility: hidden;
  }
`;

const summaryRegularStyles = `
  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
    display: none;
    visibility: hidden;
  }

  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    display: none;
    visibility: hidden;
  }
`;

const summaryStyles = {
  top: summaryTopStoryStyles,
  regular: summaryRegularStyles,
  topStories: summaryRegularStyles,
  relatedContent: summaryRegularStyles,
  leading: summaryRegularStyles,
};

export const Summary = styled.p`
  ${({ script }) => script && getLongPrimer(script)};
  ${({ service }) => getSansRegular(service)}
  color: ${C_SHADOW};
  margin: 0; /* Reset */
  padding-bottom: ${GEL_SPACING};

  ${({ promoHasImage }) => !promoHasImage && `padding-top: ${GEL_SPACING};`}

  ${({ promoType }) => summaryStyles[promoType]}
`;

Summary.propTypes = {
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
  promoHasImage: bool,
  promoType: PROMO_TYPES,
};

Summary.defaultProps = {
  promoHasImage: true,
  promoType: 'regular',
};

export const Link = styled.a`
  position: static;
  text-decoration: none;
  overflow-wrap: break-word;

  &:before {
    bottom: 0;
    content: '';
    left: 0;
    overflow: hidden;
    position: absolute;
    right: 0;
    top: 0;
    white-space: nowrap;
    z-index: 1;
  }

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  &:visited {
    color: ${C_METAL};
  }
  ${({ promoType }) => `${headlineStyle[promoType]}`}
`;

const StoryPromo = ({
  image,
  info,
  promoType,
  dir,
  displayImage,
  displayTimestamp,
  mediaIndicator,
  mediaIndicatorIsInline,
  ...props
}) => {
  const renderImage = displayImage && (
    <ImageGridItem dir={dir} promoType={promoType}>
      <ImageContentsWrapper>
        {image}
        {mediaIndicator && (
          <ImageOverlayWrapper
            mediaIndicatorIsInline={mediaIndicatorIsInline}
            promoType={promoType}
          >
            {mediaIndicator}
          </ImageOverlayWrapper>
        )}
      </ImageContentsWrapper>
    </ImageGridItem>
  );

  const renderText = (
    <TextGridItem promoType={promoType} dir={dir} displayImage={displayImage}>
      {!displayImage && mediaIndicator}
      {info}
    </TextGridItem>
  );

  return (
    <StoryPromoWrapper
      promoType={promoType}
      displayImage={displayImage}
      displayTimestamp={displayTimestamp}
      {...props}
    >
      {promoType === 'leading' ? (
        <>
          {renderText}
          {renderImage}
        </>
      ) : (
        <>
          {renderImage}
          {renderText}
        </>
      )}
    </StoryPromoWrapper>
  );
};

StoryPromo.propTypes = {
  dir: oneOf(['ltr', 'rtl']),
  image: node.isRequired,
  info: node.isRequired,
  promoType: PROMO_TYPES,
  displayImage: bool,
  displayTimestamp: bool,
  mediaIndicator: node,
  mediaIndicatorIsInline: bool,
};

StoryPromo.defaultProps = {
  dir: 'ltr',
  promoType: 'regular',
  displayImage: true,
  displayTimestamp: true,
  mediaIndicator: null,
  mediaIndicatorIsInline: false,
};

export default StoryPromo;
