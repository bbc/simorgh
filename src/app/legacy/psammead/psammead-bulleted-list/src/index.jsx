import styled from '@emotion/styled';
import { string, oneOf, shape } from 'prop-types';
import { getBodyCopy } from '#psammead/gel-foundations/src/typography';
import { getSansRegular } from '#psammead/psammead-styles/src/font-styles';
import { GEL_SPACING_DBL } from '#psammead/gel-foundations/src/spacings';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';
import { SHADOW } from '#app/components/ThemeProvider/palette';

const BulletedList = styled.ul`
  ${({ script }) => script && getBodyCopy(script)};
  ${({ service }) => getSansRegular(service)};
  margin-top: 0;
  list-style-type: none;

  & > li {
    position: relative;
    color: ${({ theme }) => theme.isDarkUi && theme.palette.GREY_2};
  }

  & > li::before {
    top: 0.5rem;
    content: ' ';
    position: absolute;
    border-width: 1rem;
    border: 0.1875rem solid
      ${({ bulletPointColour, theme }) =>
        theme.isDarkUi ? theme.palette.GREY_4 : bulletPointColour};
    background-color: ${({ bulletPointColour, theme }) =>
      theme.isDarkUi ? theme.palette.GREY_4 : bulletPointColour};
    border-radius: ${({ bulletPointShape }) =>
      bulletPointShape === 'round' ? '50%' : '0'};
    ${({ dir }) => (dir === 'rtl' ? 'right: -1rem;' : 'left: -1rem;')}
  }
`;

BulletedList.propTypes = {
  script: shape(scriptPropType).isRequired,
  dir: oneOf(['ltr', 'rtl']),
  service: string.isRequired,
  bulletPointShape: oneOf(['round', 'square']),
  bulletPointColour: string,
};

BulletedList.defaultProps = {
  dir: 'ltr',
  role: 'list',
  bulletPointShape: 'round',
  bulletPointColour: SHADOW,
};

export const BulletedListItem = styled.li`
  margin-bottom: ${GEL_SPACING_DBL};
`;

BulletedListItem.defaultProps = {
  role: 'listitem',
};

export default BulletedList;
