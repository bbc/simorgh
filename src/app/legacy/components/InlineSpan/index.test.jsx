import React from 'react';
import { shouldMatchSnapshot } from '#psammead/psammead-test-helpers/src';
import latin from '../../../components/ThemeProvider/fontScripts/latin';
import arabic from '../../../components/ThemeProvider/fontScripts/arabic';
import Inline from './index';

describe('Inline', () => {
  shouldMatchSnapshot(
    'english inline span in persian service article',
    <Inline lang="en" script={arabic} service="persian">
      This is text in a Inline.
    </Inline>,
  );

  shouldMatchSnapshot(
    'persian inline span in english service article',
    <Inline lang="fa" script={latin} service="news">
      بعض محتوى النص
    </Inline>,
  );
});
