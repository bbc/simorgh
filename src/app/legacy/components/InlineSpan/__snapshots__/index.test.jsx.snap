// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline english inline span in persian service article 1`] = `
.emotion-0 {
  padding-bottom: 1.5rem;
  margin: 0;
}

<div>
  <span
    class="emotion-0 emotion-1"
    lang="en"
  >
    This is text in a Inline.
  </span>
</div>
`;

exports[`Inline persian inline span in english service article 1`] = `
.emotion-0 {
  padding-bottom: 1.5rem;
  margin: 0;
}

<div>
  <span
    class="emotion-0 emotion-1"
    lang="fa"
  >
    بعض محتوى النص
  </span>
</div>
`;
